sklearn/feature_selection/tests/test_mutual_info.py

```diff
@@ -164,9 +173,10 @@ def test_mutual_info_classif_mixed():
         assert mi_nn[2] == mi[2]
 
 
-def test_mutual_info_options():
-    X = np.array([[0, 0, 0], [1, 1, 0], [2, 0, 1], [2, 0, 1], [2, 0, 1]], dtype=float)
-    y = np.array([0, 1, 2, 2, 1], dtype=float)
+@pytest.mark.parametrize("dtype", DTYPES)
+def test_mutual_info_options(dtype):
+    X = np.array([[0, 0, 0], [1, 1, 0], [2, 0, 1], [2, 0, 1], [2, 0, 1]], dtype=dtype)
+    y = np.array([0, 1, 2, 2, 1], dtype=dtype)
```

Comment:
I think y should be casted to a float with `dtype` only when `mutual_info is mutual_info_regression` below.

It does not really make sense to test `mutual_info_classif` on float targets. In the futures we might even want to deprecate this as it could be the cause of confusing bugs.